src/resolution/snapshot.rs

```diff
@@ -268,6 +268,55 @@ impl NpmResolutionSnapshot {
     }
   }
 
+  pub fn subset(&self, package_reqs: &[PackageReq]) -> Option<Self> {
```

Comment:
Maybe this should just panic instead of returning an `Option`?

Reply:
I think we should return an empty self?